magpie/api/management/register/register_utils.py

```diff
@@ -79,3 +139,219 @@ def get_discoverable_group_by_name(group_name, db_session):
+    # fetch user to retrieve auto-generated fields (i.e.: id)
+    tmp_user = ax.evaluate_call(
+        lambda: UserSearchService.by_user_name(user_name, status=UserStatuses.Pending, db_session=request.db),
+        http_error=HTTPForbidden, msg_on_fail=s.UserNew_POST_ForbiddenResponseSchema.description)
+
+    LOGGER.debug("[User Registration - Step 2] sending confirmation email for its validation")
+    confirmation_url = generate_callback_url(TokenOperation.USER_REGISTRATION_CONFIRM_EMAIL, request.db, user=tmp_user)
+    admin_approve = asbool(get_constant("MAGPIE_USER_REGISTRATION_APPROVAL_ENABLED", request, default_value=False,
+                                        print_missing=True, raise_missing=False, raise_not_set=False))
+    params = {
+        "user": tmp_user,
+        "confirm_url": confirmation_url,
+        "approval_required": admin_approve,
+    }
+    template = get_email_template("MAGPIE_USER_REGISTRATION_SUBMISSION_EMAIL_TEMPLATE", request)
+    ax.evaluate_call(lambda: send_email(tmp_user.email, request, template, params),
+                     fallback=lambda: request.db.rollback(), http_error=HTTPInternalServerError,
+                     msg_on_fail="Error occurred during user registration when trying to send "
+                                 "email to pending user for confirmation of its submitted email.")
+
+    user_content = rf.format_pending_user(tmp_user)
+    return ax.valid_http(http_success=HTTPCreated, detail=s.Users_POST_CreatedResponseSchema.description,
+                         content={"registration": user_content})
+
+
+def handle_user_registration_confirmation(tmp_token, request):
+    # type: (TemporaryToken, Request) -> Response
+    """
+    Applies the appropriate step of the user registration workflow following reception of the confirmation URL visit.
+
+    Implements steps (3A) and (3B) redirection of the user registration procedure.
+    Generates the appropriate response that will be displayed to the :term:`Pending User` that confirmed its email.
+
+    .. seealso::
+        - See :ref:`proc_user_registration` for the procedure step details.
+        - See :ref:`request_admin_approval` for step 3B.
+        - See :ref:`complete_user_registration` for step 5 (from 3A).
+    """
+    require_approve = asbool(get_constant("MAGPIE_USER_REGISTRATION_APPROVAL_ENABLED", request, default_value=False,
+                                          print_missing=True, raise_missing=False, raise_not_set=False))
+    if not require_approve:
+        LOGGER.debug("[User Registration - Step 3A] moving on to complete registration (no approval required)")
+        complete_user_registration(tmp_token, request)
+        data = {
+            # from the message UI page template, it is possible to see the 'Login' button in the Magpie header
+            # user should be able to find its way there to use its registered account
+            "message": cleandoc("""
```

Comment:
Why the message is not part of the mako template?

Reply:
The page is reused to render different messages based on context, potentially also for other things than user registration (eg T&C "received" indication).
Only needs to provide the value of the message.